src/trpc/routers/stakeholder-router/procedures/add-stakeholders.ts

```diff
@@ -0,0 +1,55 @@
+import { withAuth } from "@/trpc/api/trpc";
+import { ZodAddStakeholderArrayMutationSchema } from "../schema";
+
+export const addStakeholdersProcedure = withAuth
+  .input(ZodAddStakeholderArrayMutationSchema)
+  .mutation(async ({ ctx, input }) => {
+    try {
+      const companyId = ctx.session.user.companyId;
+
+      // insert companyId in every input
+      const inputDataWithCompanyId = input.map((stakeholder) => ({
+        ...stakeholder,
+        companyId,
+      }));
+
+      // get existing stakeholders
+      const existingStakeholders = await ctx.db.stakeholder.findMany({
+        where: {
+          companyId,
+        },
+        select: {
+          email: true,
+        },
+      });
+
+      // filter out stakeholders who are already in the company
+      const uniqueStakeholders = inputDataWithCompanyId.filter(
+        (stakeholder) => {
+          return !existingStakeholders.some(
+            (existingStakeholder) =>
+              existingStakeholder.email === stakeholder.email,
+          );
+        },
+      );
+
+      // bulk create
+      if (uniqueStakeholders.length > 0) {
+        await ctx.db.stakeholder.createMany({
+          data: uniqueStakeholders,
+          skipDuplicates: true,
```

Comment:
Do we need to pre-filter if we have `skipDuplicates`?

Reply:
oh yeah, even if I wrote the `skipDuplicates: true` it was still creating duplicate entries, so I wrote my own filtering.
now that I just released that the `email` field in the schema should be `@unique` which wasn't before.
I added a unique constraint in the email field in the schema and now it is working 👍 

Thanks !